ccmlib/scylla_docker_cluster.py

```diff
@@ -0,0 +1,208 @@
+        return super(ScyllaDockerNode, self).start(*args, **kwargs)
+
+    def _start_scylla(self, args, marks, update_pid, wait_other_notice,
+                      wait_for_binary_proto, ext_env):
+        # TODO: handle smp correctly via the correct param/api (or only via commandline params)
+        # TODO: mount of the data dir
+        # TODO: pass down the full command line params, since the docker ones doesn't support all of them ?
+        # TODO: pass down a unique tag, with the cluster name, or id, if we have such in ccm, like test_id in SCT ?
+
+        rc, out, err = self.create_docker()
+        if not rc:
+            raise BaseException(f'failed to create docker {self.docker_name}')
+
+        time.sleep(5)
+        scylla_status = self.service_status('scylla')
+        if scylla_status and scylla_status.upper() != 'RUNNING':
+            self.service_start('scylla')
+
+    def do_stop(self, gently=True):
+        """
+        Stop the node.
+          - gently: Let Scylla and Scylla JMX clean up and shut down properly.
+            Otherwise do a 'kill -9' which shuts down faster.
+        """
+        if gently:
+            self.service_stop('scylla-jmx')
+            self.service_stop('scylla')
+        else:
+            raise NotImplementedError()
+
+    def clear(self, *args, **kwargs):
+        res = run(['bash', '-c', f'docker rm -f {self.pid}'], stdout=PIPE, stderr=PIPE)
+        self.log_thread.stop()
+        super(ScyllaDockerNode, self).clear(*args, **kwargs)
+
+    def _start_jmx(self, data):
+        jmx_status = self.service_status('scylla-jmx')
+        if not jmx_status and jmx_status.upper() == 'RUNNING':
+            self.service_start('scylla-jmx')
+
+    def is_running(self):
+        """
+        Return true if the node is running
+        """
+        self.__update_status()
+        return self.status == Status.UP or self.status == Status.DECOMMISSIONED
+
+    def is_live(self):
+        """
+        Return true if the node is live (it's run and is not decommissioned).
+        """
+        self.__update_status()
+        return self.status == Status.UP
+
+    def __update_status(self):
+        if self.pid is None:
+            if self.status == Status.UP or self.status == Status.DECOMMISSIONED:
+                self.status = Status.DOWN
+            return
+
+        old_status = self.status
+
+        scylla_status = self.service_status('scylla')
+        if scylla_status and scylla_status.upper() == 'RUNNING':
+            self.status = Status.UP
+        else:
+            self.status = Status.DOWN
+
+    def _wait_java_up(self, ip_addr, jmx_port):
+        return True
+
+    def _update_pid(self, process):
+        pass
+
+    def get_tool(self, toolname):
+        return ['docker',  'exec', '-i',  f'{self.pid}', f'{toolname}']
+
+    def get_env(self):
+        return os.environ.copy()
+
+
+import subprocess
+from threading import Thread, Event as ThreadEvent
+
+
+class DockerLogger:
```

Comment:
Add this to get the `docker logs` in real time into a file, so all the `mark_log` functionality would work